File: typescript/src/advertisements/user/domain/value-object/UserStatus.ts
export class UserStatus {
    public static readonly ENABLED = new UserStatus('enabled');
    public static readonly DISABLED = new UserStatus('disabled');

    private constructor(private readonly _value: string) {}

    public static fromString(status: string): UserStatus {
        switch (status) {
            case 'enabled':
                return UserStatus.ENABLED;
            case 'disabled':
                return UserStatus.DISABLED;
            default:
                throw new Error(`Invalid status: ${status}`);
        }
    }

    public value(): string {
        return this._value;
    }

    public isEnabled(): boolean {
        return this === UserStatus.ENABLED;
    }
}

File: kotlin/src/main/kotlin/advertisements/user/domain/value_object/UserStatus.kt
package advertisements.user.domain.value_object

enum class UserStatus {
    ENABLED,
    DISABLED;

    companion object {
        fun fromString(status: String): UserStatus {
            return when (status.lowercase()) {
                "enabled" -> ENABLED
                "disabled" -> DISABLED
                else -> throw IllegalArgumentException("Invalid status: $status")
            }
        }
    }

    fun value(): String {
        return when (this) {
            ENABLED -> "enabled"
            DISABLED -> "disabled"
        }
    }

    fun isEnabled(): Boolean {
        return this == ENABLED
    }
}
